apps/openmw/mwrender/animation.cpp

```diff
@@ -1106,7 +1183,25 @@ namespace MWRender
         int mLowestUnusedTexUnit;
     };
 
-    void Animation::addGlow(osg::ref_ptr<osg::Node> node, osg::Vec4f glowColor)
+    void Animation::addSpellCastGlow(const ESM::MagicEffect *effect){
+
+        osg::Vec4f glowColor(1,1,1,1);
+        glowColor.x() = effect->mData.mRed / 255.f;
+        glowColor.y() = effect->mData.mGreen / 255.f;
+        glowColor.z() = effect->mData.mBlue / 255.f;
+
+        if (!mObjectRoot->getUpdateCallback()) // If there is no glow on object
+            addGlow(mObjectRoot, glowColor, 1.5); // Glow length measured from in-game as about 1.5 seconds
+
+        else if (dynamic_cast <GlowUpdater*>(mObjectRoot->getUpdateCallback())->isDone() == true) // If there was a temporary glow on object and it finished
```

Comment:
If you use a dynamic_cast you need to check the result of the cast.

```
A* a = dynamic_cast<A*>(b);
if (a)
    // do stuff
```

or

```
A& a = dynamic_cast<A&>(*b); // throws an exception if the cast fails
```

If you are sure that the cast will not fail you should use a static_cast.

In this case I would prefer storing a pointer to the SpellGlowUpdater in ```this``` to eliminate the need for casting.